Issue/2896 owner_lifecycle_state

Imat00 opened the pull request:
### Changes
- Added new table owner_lifecycle_state and linked it to owner
- Added a dropdown for selecting lifecycle state in "Add Owner" and "Edit Owner" forms
- Created Managemenu for lifecycles in settings - owner (create/edit/delete)

### Related Issue

GUI and Database: add lifecycle meta field for owner/application #2896
Resolves #2896

Imat00 commented:
Clean install and upgrade on local VM was successfull

tpurschke reviewed (changes requested):
Functionality is perfect, 

I see just wording issues here:
- please change all names (including DB ids) so that "state" is always included.
- a
  - ID lifecyclestate --> owner_lc_state
  - Produktionszyklus --> Produktionsstatus 
  - Lifecycle State --> Production state
- b
  - DB ID lifecycles --> owner_lc_states
  - Produktionszyklen --> Produktions-Stati
  - Lifecycles --> Production Lifecycle States
- c
  - ID manage_lifecycles --> manage_owner_lc_state
  - Produktionszyklen --> Produktions-Lifecylce-Systati ändern
  - Manage Lifecycles --> Manage production lifecycle states
- d
  - DB ID add_lifecycle --> add_owner_lc_state
  - Add lifecycle --> Add Owner Lifecycle State
  - Produktionszyklus --> Lifecycle Status hinzufügen
  
...

Imat00 commented:
> wording issues

changed texts to and tried follow your example:

id:   edit_owner_lc_state
german:  Lifecycle Status bearbeiten
english:   Edit Owner Lifecycle State

id:   delete_owner_lc_state
german:  Lifecycle Status löschen
english:   Delete Owner Lifecycle State

Tooltips:
Before
After

Sind sie sicher, dass sie folgenden Produktionszyklus l&ouml;schen wollen: 
Sind sie sicher, dass sie folgenden Lifecycle Status l&ouml;schen wollen: 

Are you sure you want to delete lifecycle: 
Are you sure you want to delete Owner Lifecycle State: 

Produktionszyklus konnte nicht gespeichert werden
Lifecycle Status konnte nicht gespeichert werden

Lifecycle could not be saved
Owner Lifecycle State could not be saved

Produktionszyklus: Gibt an, in welchem Produktionszyklus sich der Eigentümer befindet.
Produktionsstatus: Gibt an, in welchem Lifecycle Status sich der Eigentümer befindet.

Lifecycle State: The lifecycle phase assigned to the owner.
Production state: The Owner Lifecycle State assigned to the owner.

Name: Name des Produktionszyklus
Name: Name des Produktionsstatus

Name: Lifecycle Status name
Name: Owner Lifecycle State name

In diesem Kapitel können die Produktionszyklen von Eigent&uuml;mern eingesehen, erstellt, bearbeitet und gelöscht werden. Produktionszyklen definieren den aktuellen Zustand eines Eigent&uuml;mers (z. B. Geplant, Im Betrieb, Auslaufend).
In diesem Kapitel können die Produktions-Stati von Eigent&uuml;mern eingesehen, erstellt, bearbeitet und gelöscht werden. Produktions-Stati definieren den aktuellen Zustand eines Eigent&uuml;mers (z. B. Geplant, Im Betrieb, Auslaufend).

In this chapter, the lifecycle states of owners can be viewed, created, edited, and deleted. Lifecycle states define the current status of an owner (e.g., Planned, In Production, End of Life).
In this chapter, Production Lifecycle States of owners can be viewed, created, edited, and deleted. Production Lifecycle States define the current status of an owner (e.g., Planned, In Production, End of Life).

tpurschke commented on roles/database/files/sql/idempotent/fworch-texts.sql:
```diff
@@ -2402,6 +2404,16 @@ INSERT INTO txt VALUES ('assign_state_to',      'German',   'Status zuordnen zu:
 INSERT INTO txt VALUES ('assign_state_to',      'English',  'Assign state to: ');
 INSERT INTO txt VALUES ('select_action',        'German',   'Aktion ausw&auml;hlen');
 INSERT INTO txt VALUES ('select_action',        'English',  'Select action');
+INSERT INTO txt VALUES ('manage_owner_lc_state','German',   'Produktions-Lifecylce-Systati ändern');
```
typo

Imat00 commented on roles/database/files/sql/idempotent/fworch-texts.sql:
```diff
@@ -2402,6 +2404,16 @@ INSERT INTO txt VALUES ('assign_state_to',      'German',   'Status zuordnen zu:
 INSERT INTO txt VALUES ('assign_state_to',      'English',  'Assign state to: ');
 INSERT INTO txt VALUES ('select_action',        'German',   'Aktion ausw&auml;hlen');
 INSERT INTO txt VALUES ('select_action',        'English',  'Select action');
+INSERT INTO txt VALUES ('manage_owner_lc_state','German',   'Produktions-Lifecylce-Systati ändern');
```
changed to 'Produktions-Lifecylce-Stati ändern'

tpurschke reviewed (approved):
excellent
